src/app/child-dev-project/children/child-details/child-details.component.spec.ts

```diff
@@ -1,91 +1,103 @@
 import { async, ComponentFixture, TestBed } from "@angular/core/testing";
-
-import { MatDialog } from "@angular/material/dialog";
-import { ActivatedRoute, Router } from "@angular/router";
-import { FormBuilder } from "@angular/forms";
-import { EntityMapperService } from "../../../core/entity/entity-mapper.service";
-import { AlertService } from "../../../core/alerts/alert.service";
-import { ConfirmationDialogService } from "../../../core/confirmation-dialog/confirmation-dialog.service";
-import { DatePipe, Location, PercentPipe } from "@angular/common";
-import { of } from "rxjs";
 import { ChildDetailsComponent } from "./child-details.component";
-import { MatSnackBar } from "@angular/material/snack-bar";
-import { databaseServiceProvider } from "app/core/database/database.service.provider";
-import { SessionService } from "app/core/session/session-service/session.service";
-import { EntitySchemaService } from "../../../core/entity/schema/entity-schema.service";
-import { MockDatabase } from "app/core/database/mock-database";
-import { SchoolsService } from "app/child-dev-project/schools/schools.service";
+import { MockDatabase } from "../../../core/database/mock-database";
 import { ChildPhotoService } from "../child-photo-service/child-photo.service";
-import { ChildrenModule } from "../children.module";
-import { FormDialogModule } from "../../../core/form-dialog/form-dialog.module";
+import { Observable, of, Subscriber } from "rxjs";
+import { ActivatedRoute, Router } from "@angular/router";
+import { Location } from "@angular/common";
+import { SessionService } from "../../../core/session/session-service/session.service";
+import { User } from "../../../core/user/user";
 import { MatNativeDateModule } from "@angular/material/core";
-import { User } from "app/core/user/user";
+import { ChildrenModule } from "../children.module";
+import { databaseServiceProvider } from "../../../core/database/database.service.provider";
+import { Child } from "../model/child";
+import { EntityMapperService } from "../../../core/entity/entity-mapper.service";
+import { ChildrenService } from "../children.service";
 
 describe("ChildDetailsComponent", () => {
   let component: ChildDetailsComponent;
   let fixture: ComponentFixture<ChildDetailsComponent>;
+
+  let routeObserver: Subscriber<any>;
   const mockedRoute = {
-    paramMap: of({ get: () => "new" }),
-    data: of({ icon: "child" }),
+    paramMap: new Observable((observer) => {
+      routeObserver = observer;
+      observer.next({ get: () => "new" });
+    }),
+    data: of({
+      panels: [
```

Comment:
if we don't have specific unit tests requiring all the panels, we can reduce this here in the spec to a minimal config, maybe just with one Form and one special component panel